x-pack/plugin/logsdb/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/TextRollingUpgradeIT.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.logsdb;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class TextRollingUpgradeIT extends AbstractLogsdbRollingUpgradeTestCase {
+
+    private static final String DATA_STREAM_NAME = "logs-text-bwc-test";
+
+    private static final List<String> FIELD_VALUES_1 = Arrays.asList(
+        "the quick brown fox jumps over the lazy dog",
+        "elasticsearch is a distributed search engine",
+        "logsdb provides efficient log storage",
+        "synthetic source reduces storage costs",
+        "text fields support full text search"
+    );
+
+    private static final List<String> FIELD_VALUES_2 = Arrays.asList(
+        "rolling upgrades test cluster compatibility",
+        "phrase queries match exact word sequences",
+        "text analysis tokenizes content for indexing",
+        "match queries find relevant documents",
+        "search across upgraded and non-upgraded nodes",
+        "cluster health remains green during upgrades"
+    );
+
+    private static final String ITEM_TEMPLATE = """
+        {"@timestamp": "$now", "message": "$message"}
+        """;
+
+    private static final String TEMPLATE = """
+        {
+            "mappings": {
+              "properties": {
+                "@timestamp" : {
+                  "type": "date"
+                },
+                "message": {
+                  "type": "text"
+                }
+              }
+            }
+        }""";
+
+    public void testIndexingWithTextField() throws Exception {
+        beforeUpgrade();
+        upgradeAllNodes();
```

Comment:
I think we should upgrade one node at a time here. After each node upgrade we should do checks like in `afterUpgrade(...)`. This way we get more rolling upgrade test coverage than just upgrading all nodes at once.